lua/neogit/spinner.lua

```diff
@@ -0,0 +1,67 @@
+---@class Spinner
+---@field text string
+---@field count number
+---@field interval number
+---@field pattern string[]
+---@field timer uv_timer_t
+---@field start fun(self)
+---@field stop fun(self)
+local Spinner = {}
+Spinner.__index = Spinner
+
+---@return Spinner
+function Spinner.new(text)
+  local instance = {
+    text = text,
+    interval = 100,
+    count = 0,
+    timer = nil,
+    pattern = {
+      "⠋",
+      "⠙",
+      "⠹",
+      "⠸",
+      "⠼",
+      "⠴",
+      "⠦",
+      "⠧",
+      "⠇",
+      "⠏",
+    },
```

Comment:
Making this customizable would be perfect :) here are some ideas if you'd like inspiration https://github.com/gtramontina/spinner-frames.cr/blob/master/src/spinner-frames/charset.cr

Reply:
sure :D